src/mozillavpn.cpp

```diff
@@ -298,7 +298,6 @@ void MozillaVPN::initialize() {
   AndroidUtils::instance();
 #endif
 
-  m_private->m_captivePortalDetection.initialize();
```

Comment:
Should all the CaptivePortalDetection related code be removed?

Reply:
Hmm, are you referring to the following 3?

```
  connect(&m_private->m_connectionManager, &ConnectionManager::stateChanged,
          &m_private->m_captivePortalDetection,
          &CaptivePortalDetection::stateChanged);

  connect(&m_private->m_connectionHealth, &ConnectionHealth::stabilityChanged,
          &m_private->m_captivePortalDetection,
          &CaptivePortalDetection::stateChanged);

  connect(SettingsHolder::instance(),
          &SettingsHolder::captivePortalAlertChanged,
          &m_private->m_captivePortalDetection,
          &CaptivePortalDetection::settingsChanged);
```

Reply:
Yes, and the CaptivePortalDetection class.

Reply:
Never mind, we still need these based on the PR description.

Reply:
However, can these 3 be moved into CaptivePortalDetection.initialize() and remove the call to set up a slot for NetworkWatcher::networkChange? This call can be restored then, which seems cleaner than not calling m_captivePortalDetection.initialize when CaptivePortalDetection is still used.

Reply:
For my understanding, could you please explain why this line was removed? It seems like CaptivePortalDetection still needs the code that is currently in m_captivePortalDetection.initialize().

Reply:
IIUC, this is the meaty change that intends to disable the captive portal, but I don't think this is sufficient, digging into the code we find in `src/captiveportal/captiveportaldetection.cpp`:
```
void CaptivePortalDetection::initialize() {
  m_active = SettingsHolder::instance()->captivePortalAlert();
  const auto networkWatcher = MozillaVPN::instance()->networkWatcher();
  connect(networkWatcher, &NetworkWatcher::networkChange, this,
          &CaptivePortalDetection::networkChanged);
}
```

What I think we really desire is the removal of the `CaptivePortalMonitor` class which generates periodic checks.